core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestApplyTableScanRedirection.java

```diff
@@ -227,28 +225,6 @@ public void testApplyTableScanRedirectionWithFilter()
                                             equalTo(new MockConnectorTableHandle(destinationTable)),
                                             TupleDomain.all(),
                                             ImmutableMap.of("DEST_COL", equalTo(destinationColumnHandleA)))));
-
-            ruleTester.assertThat(applyTableScanRedirection)
-                    .on(p -> {
-                        Symbol column = p.symbol(sourceColumnNameB, VARCHAR);
-                        return p.tableScan(
-                                createTableHandle(new MockConnectorTableHandle(sourceTable, constraint, Optional.empty())),
-                                ImmutableList.of(column),
-                                ImmutableMap.of(column, sourceColumnHandleB), // predicate on non-projected column
-                                constraint);
```

Comment:
This test was meant to simulate the following scenario:
There are 2 columns A and B, there is a predicate on A and projection on B
Constraint on column A is pushed into TS and enforced by connector
Column A assignment is pruned from TS by column pruning rule

Maybe instead of removing, the test case can be fixed by passing `TupleDomain.all` in enforcedConstraint (or whatever is left in enforcedConstraint after PPD when a constraint is enforced by connector and engine does not need to do any extra work). I think we are correctly sending `constraint` into `MockConnectorTableHandle` to mimc PPD.

Reply:
@raunaqmorarka is that test case covered by some test in `TestTableScanRedirectionWithPushdown`?

Reply:
`TestTableScanRedirectionWithPushdown#testRedirectAfterColumnPruningOnPushedDownPredicate` is also about this scenario. However, there we are testing the end result of applying multiple rules. We are not explicitly verifying the result of just running the redirection rule in this scenario elsewhere.
We are preserving predicate on A in the mock table handle. I'm assuming this is sufficient to mimic the PPD part of the above scenario and that the TupleDomain in enforced constraint doesn't need to have predicate on A.

Reply:
From unit test perspective, this will be equivalent to unfiltered case, covered by `io.trino.sql.planner.iterative.rule.TestApplyTableScanRedirection#testApplyTableScanRedirection`.

Reply:
It wouldn't be equivalent to unfiltered case because the `constraint` would still be passed to `MockConnectorTableHandle` and the test will continue to match for a `filter` node.

Reply:
@raunaqmorarka right, thanks, fixed.